src/index/hnsw/impl/BitsetFilter.h

```diff
@@ -0,0 +1,35 @@
+// Copyright (C) 2019-2024 Zilliz. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under the License.
+
+#pragma once
+
+#include <faiss/MetricType.h>
+
+#include "knowhere/bitsetview.h"
+
+namespace knowhere {
+
+// specialized override for knowhere
+struct BitsetFilter {
```

Comment:
A noob Q. Why we need `BitsetFilter` and `DefaultIDSelectorFilter` instead of using `Selector` directly

Reply:
`Selector` is based on `virtual` methods. Technically, I haven't checked whether a compiler de-virtualized such calls. I need to check this.

Reply:
updated. I had to use a small hack, because `GCC` seems to have problems devirtualizing. There were no problems with `clang`.